gateway/gateway.go

```diff
@@ -306,51 +282,28 @@ func (g *gateway) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 		return fmt.Errorf("error initializing the gateway: %w", err)
 	}
 
-	region := getRegion(cfg.Region)
+	region := lorahw.GetRegion(cfg.Region)
 	switch region {
-	case US, unspecified:
+	case lorahw.US, lorahw.Unspecified:
 		g.logger.Infof("configuring gateway for US915 band")
 		g.regionInfo = regionInfoUS
-	case EU:
+	case lorahw.EU:
 		g.logger.Infof("configuring gateway for EU868 band")
 		g.regionInfo = regionInfoEU
-
 	}
-	errCode := C.setUpGateway(C.int(cfg.Bus), C.int(region))
-	if errCode != 0 {
-		strErr := parseErrorCode(int(errCode))
-		return fmt.Errorf("failed to set up the gateway: %s", strErr)
+
+	if err := lorahw.SetupGateway(cfg.Bus, region); err != nil {
+		return fmt.Errorf("failed to set up the gateway: %w", err)
```

Comment:
Similar comment, not a big fan of region logic getting split between the packages.